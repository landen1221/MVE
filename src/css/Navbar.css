.MuiTypography-root {
  font-weight: bold;
  letter-spacing: 0.05em;
  font-family: Roboto, Helvetica, Arial, sans-serif;
}

.makeStyles-search-1 {
  border-radius: 12px;
}

.MuiButton-root {
  color: white;
  border-radius: 15px;
}

#logo {
  color: white;
}

.Navbar #logo-small {
  color: white;
}

.Navbar #logo-link {
  text-decoration: none;
}

.Navbar button {
  background-color: rgba(255, 255, 255, 0);
  color: white;
}

.Navbar .MuiButton-outlined {
  border: 1px solid #fff;
}

.MuiBox-root-5 {
  border: none;
  padding-top: 10px;
  padding-bottom: 0;
  margin-bottom: 0;
}

#add-icon {
  font-size: 5em;
}

#nav-grid {
  text-align: center;
}

#center-me {
  text-align: center;
  align-content: center;
  max-width: 400px;
}

.Navbar button {
  margin-left: auto;
}

#search-area {
  max-width: 25vw;
  min-width: 120px;
}

.Navbar .MuiToolbar-gutters {
  margin-left: auto;
  margin-right: auto;
  padding-left: 10px;
}

/* #toolbar {
  max-width: 45vw;
} */
