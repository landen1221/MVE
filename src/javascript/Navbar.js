import React, { useState } from "react";
import { fade, makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import SearchIcon from "@material-ui/icons/Search";
import Button from "@material-ui/core/Button";
import "../css/Navbar.css";
import { Link, useHistory } from "react-router-dom";
import Hidden from "@material-ui/core/Hidden";
import Grid from "@material-ui/core/Grid";

const useStyles = makeStyles((theme) => ({
  search: {
    backgroundColor: fade(theme.palette.common.white, 0.15),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    position: "relative",
    borderRadius: theme.shape.borderRadius,

    [theme.breakpoints.down("sm")]: {
      marginRight: theme.spacing(3),
      marginLeft: theme.spacing(3),
      width: "80%",
    },

    [theme.breakpoints.up("sm")]: {
      marginRight: theme.spacing(10),
      marginLeft: theme.spacing(10),
      width: "35%",
    },
    [theme.breakpoints.up("lg")]: {
      marginRight: theme.spacing(30),
      marginLeft: theme.spacing(30),
      width: "25%",
    },
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  inputRoot: {
    color: "inherit",
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create("width"),
  },
  toolbar: {
    [theme.breakpoints.down("sm")]: {
      width: "100%",
    },

    [theme.breakpoints.up("sm")]: {
      width: "85%",
    },
    [theme.breakpoints.up("lg")]: {
      width: "65%",
      maxWidth: "2000px",
    },
  },
}));

function Navbar({ setSearchBy }) {
  let history = useHistory();
  const classes = useStyles();

  const [searchTerm, setSearchTerm] = useState({ search: "" });

  // example search url: myvaccineexprience.org/story/search?q=worth+it
  function handleSubmit(evt) {
    if (evt.key === "Enter") {
      let finalSearch = searchTerm.search.split(" ").join("+");
      setSearchBy(finalSearch);
      history.push(`/story/search`);
    }
  }

  function handleChange(evt) {
    const { name, value } = evt.target;
    setSearchTerm((f) => ({
      ...f,
      [name]: value,
    }));
  }
  const { innerWidth } = window;
  return (
    <div className={`Navbar`}>
      <Grid container spacing={1} className="navbar-grid">
        <AppBar position="fixed" id="AppBar">
          <Toolbar className={classes.toolbar} id="toolbar">
            <Grid item xs={3}>
              <div className="logo">
                <Link to="/" id="logo-link">
                  <Hidden smDown>
                    <Typography variant="h6" noWrap id="logo">
                      MyVaccineExperience.org
                    </Typography>
                  </Hidden>
                  <Hidden mdUp>
                    <Typography variant="inherit" noWrap id="logo-small">
                      MVE
                    </Typography>
                  </Hidden>
                </Link>
              </div>
            </Grid>
            <Grid item xs={6} id="grid-item-search">
              <div className={classes.search} id="search-area">
                <div className={classes.searchIcon}>
                  <SearchIcon />
                </div>
                <InputBase
                  placeholder={innerWidth > 400 ? "Search..." : "..."}
                  name="search"
                  value={searchTerm.search}
                  onChange={handleChange}
                  onKeyPress={handleSubmit}
                  classes={{
                    root: classes.inputRoot,
                    input: classes.inputInput,
                  }}
                  inputProps={{ "aria-label": "search" }}
                />
              </div>
            </Grid>
            <Grid item xs={3}>
              <div className="story-button">
                <Link to="/add-story">
                  <Button
                    variant="outlined"
                    color="default"
                    className="add-story"
                  >
                    <Hidden mdDown>Add Story </Hidden>
                    <Hidden lgUp id="add-icon">
                      New
                    </Hidden>
                  </Button>
                </Link>
              </div>
            </Grid>
          </Toolbar>
        </AppBar>
      </Grid>
    </div>
  );
}

export default Navbar;
