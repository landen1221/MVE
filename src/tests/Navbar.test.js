import ReactDOM from "react-dom";
import Navbar from "../javascript/Navbar";
import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import userEvent from "@testing-library/user-event";

// Props: setSearchBy

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <MemoryRouter>
      <Navbar></Navbar>
    </MemoryRouter>,
    div
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <Navbar></Navbar>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("searchbar functioning", () => {
  const mockOnSubmit = jest.fn();
  render(
    <MemoryRouter>
      <Navbar setSearchBy={mockOnSubmit}></Navbar>
    </MemoryRouter>
  );

  // test that I can type into search
  const searchBar = screen.getByPlaceholderText("Search...");
  userEvent.type(searchBar, "Test Search");
  expect(searchBar).toHaveValue("Test Search");

  // test that enter fires
  userEvent.keyboard(`{enter}`);
  expect(mockOnSubmit).toHaveBeenCalled();
});

test("Logo references homepage", () => {
  render(
    <MemoryRouter>
      <Navbar></Navbar>
    </MemoryRouter>
  );
  const logo = screen.getByText("MyVaccineExperience.org");
  expect(logo.parentElement).toHaveAttribute("href", "/");
});

test("Add Story Button references /add-story", () => {
  render(
    <MemoryRouter>
      <Navbar></Navbar>
    </MemoryRouter>
  );
  const addStory = screen.getByText("ADD STORY", { exact: false });
  expect(addStory.parentElement.parentElement).toHaveAttribute(
    "href",
    "/add-story"
  );
});
