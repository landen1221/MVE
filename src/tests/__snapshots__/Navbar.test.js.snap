// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navbar undefined"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          href="/"
        >
          <span
            class="MuiTypography-root MuiTypography-noWrap"
            id="logo"
          >
            MyVaccineExperience.org
          </span>
        </a>
        <div
          class="makeStyles-search-1"
        >
          <div
            class="makeStyles-searchIcon-2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root makeStyles-inputRoot-3"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input makeStyles-inputInput-4"
              name="search"
              placeholder="Search..."
              type="text"
              value=""
            />
          </div>
        </div>
        <a
          href="/add-story"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add Story
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <div />
        <div />
        <div />
      </div>
    </header>
  </div>
</DocumentFragment>
`;
